src/run/mod.rs

```diff
@@ -38,182 +36,260 @@ impl Plugin for RunPlugin {
-        ))
-        .insert_resource(LevelsResource::from(
-            from_bytes::<LevelsResourceData>(include_bytes!("../../assets/data/levels.ron"))
+        .insert_resource(
+            from_bytes::<PremadeRunsResource>(include_bytes!("../../assets/data/premade_runs.ron"))
                 .unwrap(),
-        ));
+        )
+        .insert_resource(
+            from_bytes::<PremadeLevelsResource>(include_bytes!(
+                "../../assets/data/premade_levels.ron"
+            ))
+            .unwrap(),
+        )
+        .insert_resource(RunResource::default());
 
         app.add_event::<SpawnFormationEvent>()
             .add_event::<LevelCompletedEvent>()
             .add_event::<RunEndEvent>();
 
-        app.add_systems(
-            OnEnter(states::AppStates::Game),
-            setup_first_level.in_set(GameEnterSet::BuildLevel),
-        );
+        app.add_systems(OnEnter(AppStates::InitializeRun), init_run_system);
 
         app.add_systems(
             Update,
-            (
-                level_system.in_set(GameUpdateSet::Level),
-                spawn_formation_system.in_set(GameUpdateSet::Spawn),
-                next_level_system.in_set(GameUpdateSet::NextLevel),
-            )
-                .run_if(in_state(states::AppStates::Game))
-                .run_if(in_state(states::GameStates::Playing)),
+            (tick_run_system, handle_objective_system, run_end_system)
+                .in_set(GameUpdateSet::Level)
+                .run_if(in_state(AppStates::Game))
+                .run_if(in_state(GameStates::Playing)),
         );
 
         app.add_systems(
             Update,
-            reset_run_system.run_if(in_state(states::AppStates::GameOver)),
+            spawn_formation_system
+                .in_set(GameUpdateSet::Spawn)
+                .run_if(in_state(AppStates::Game))
+                .run_if(in_state(GameStates::Playing)),
         );
 
-        app.add_systems(
-            Update,
-            reset_run_system.run_if(in_state(states::AppStates::Victory)),
-        );
-
-        app.add_systems(
-            Update,
-            reset_run_system.run_if(in_state(states::GameStates::Paused)),
-        );
+        // reset the run after exiting the end game screens and when entering the main menu
+        app.add_systems(OnExit(AppStates::GameOver), run_reset_system);
+        app.add_systems(OnExit(AppStates::Victory), run_reset_system);
+        app.add_systems(OnEnter(AppStates::MainMenu), run_reset_system);
     }
 }
 
-pub enum RunOutcomeType {
-    #[allow(dead_code)] // TODO: Why is this never constructed????
-    Victory,
-    Defeat(RunDefeatType),
+#[derive(Resource, Deserialize)]
+pub struct PremadeRunsResource {
+    pub runs: HashMap<String, Vec<String>>,
 }
 
-pub enum RunDefeatType {
-    #[allow(dead_code)] // TODO: Why is this never constructed????
-    PlayersDestroyed,
-    DefenseDestroyed,
+#[derive(Resource, Default, Debug)]
+pub struct RunResource {
```

Comment:
rename to `CurrentRunProgress` or `CurrentRunState` just to emphasize that this thing is a state-machine of sorts.

Reply:
I'm renaming it to CurrentRunProgressResource because I want it to still be marked as a Resource.